### PR TITLE
feat(dlp): added a sample for dlp redact image all text

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [dlp redact image all text](https://cloud.google.com/dlp/docs/samples/dlp-redact-image-all-text) with a unit test case.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by annadata — APPROVED** [review]

File: dlp/snippets/redact/testdata/image.jpg
Comment (telpirion) [inline dlp/snippets/redact/testdata/image.jpg:1]: Please remove trademarked material:

+ Honda
+ Accord
+ LX

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/redact/testdata/image.jpg
Comment (annadata) [reply dlp/snippets/redact/testdata/image.jpg:1]: @telpirion Thanks for pointing this out. @AarshD-crest is just using the images from our documentation. 
@AarshD-crest you can snip out the area where this information is present or replace it with some generic words in the image to make progress.

@catherinerosesantos Please have a look at this. Do we need to update our documentation?

Comment (AarshD-crest) [reply dlp/snippets/redact/testdata/image.jpg:1]: Sure @annadata, I have updated the image as per the suggestions.

Comment (telpirion) [reply dlp/snippets/redact/testdata/image.jpg:1]: @annadata If our documentation includes this, then it should be scrubbed there, too. I would log a bug with the docs team. Google doesn't own the trademark for "Honda," "Accord," or "LX." We need to be careful about how we use other companies' trademarks.

**Review by telpirion — APPROVED** [review]

**Pull request merged**